client/src/pages/pot-instance/components/PotView/PotView.tsx

```diff
@@ -10,18 +11,39 @@ export enum PotViewState {
 
 interface Props {
   state: PotViewState;
-  potContent: PotContent[];
-  setPotContent: Dispatch<SetStateAction<PotContent[]>>;
   addToCookedPot: (foods: PotContent[]) => void;
 }
 
 // Data and timer handler for things in a pot instance.
-const PotView = ({
-  state,
-  potContent,
-  setPotContent,
-  addToCookedPot
-}: Props) => {
+const PotView = ({ state, addToCookedPot }: Props) => {
+  const [potContent, setPotContent] = useState<PotContent[]>([]);
+
+  // function takes in the item sent by the parent when an ingredients is added to the pot and updates the foodtimerObj state
+  const addFoodTimer = useCallback(
+    (
+      itemName: string,
+      itemCategory: string,
+      startTime: number,
+      finishTime: number,
+      remainingTime: number
+    ) => {
+      const tempObj = potContent;
+      let addObj = {} as PotContent;
+      addObj = {
+        id: uniqid(),
+        name: itemName,
+        category: itemCategory,
+        currentTime: startTime,
+        endTime: finishTime,
+        timeLeft: remainingTime
+      };
+      tempObj.push(addObj);
+      tempObj.sort((a, b) => a.currentTime - b.currentTime);
+      setPotContent([...tempObj]);
+    },
+    [setPotContent]
```

Comment:
How does useCallback prevent the re-render?

"It will return a memoized version of the callback that will only change if one of the dependencies has changed".
But won't all the values in the callback be different based on the ingredient that is being added? Or does it work like this:
1) You add Beef (new)
2) You add Chicken (new)
3) You add Beef (again) <- this is memoized and does not re-render?

Reply:
One thing that happens when the parent component updates any update function like addFoodTimer will turn into a different function object (same behaviour just remade so it looks different in memory). 
This then gets passed into the child component which also thinks its different due to the above effect and causes the child component to rerender on a change that isn't a change.
But memoizing the function with dependency of setPotContent which is a state hook in this case we can ensure that the function object of addFoodTimer doesn't update unless the parent component is freshly mounted.